onnx_neural_compressor/algorithms/utility.py

```diff
@@ -0,0 +1,857 @@
+    "int64": 7,
+    "string": 8,
+    "bool": 9,
+    "fp16": 10,
+    "float16": 10,
+    "double": 11,
+    "uint32": 12,
+    "uint64": 13,
+    "complex64": 14,
+    "complex128": 15,
+    "bf16": 16,
+    "bfloat16": 16,
+}
+
+QUANT_OP_NAME_SUFFIX = "_quant"
+__producer__ = "onnx.quantize"
+__version__ = "0.1.0"
+onnx_domain = "ai.onnx"
+ms_domain = "com.microsoft"
+
+ONNX_INT_TYPE_RANGE = {
+    onnx.TensorProto.UINT8: (0, 255),
+    onnx.TensorProto.INT8: (-128, 127),
+}
+
+ONNX_INT_TYPE_SYMMETRIC_RANGE = {
+    onnx.TensorProto.INT8: (-127, 127),
+}
+
+ONNX_INT_TYPE_REDUCED_RANGE = {
+    onnx.TensorProto.UINT8: (0, 127),
+    onnx.TensorProto.INT8: (-64, 64),
+}
+
+
+def check_model_with_infer_shapes(model):
+    """Check if the model has been shape inferred."""
+    if isinstance(model, (pathlib.Path, str)):
+        model = onnx.load(model, load_external_data=False)
+    elif isinstance(model, onnx_model.ONNXModel):
+        model = model.model
+    if len(model.graph.value_info) > 0:
+        return True
+    return False
+
+
+def find_by_name(name, item_list):
+    """Helper function to find item by name in a list."""
+    items = []
+    for item in item_list:
+        assert hasattr(item, "name"), "{} should have a 'name' attribute defined".format(item)  # pragma: no cover
+        if item.name == name:
+            items.append(item)
+    if len(items) > 0:
+        return items[0]
+    else:
+        return None
+
+
+def is_quantizable_type(data_type):
+    return data_type in [onnx.TensorProto.FLOAT, onnx.TensorProto.FLOAT16, onnx.TensorProto.BFLOAT16]
+
+
+def get_qmin_qmax_for_qType(qType, reduce_range=False, sym=False):  # noqa: N802
+    """Get qmin, qmax for qType."""
+    if qType == onnx.TensorProto.FLOAT8E4M3FN:
+        raise NotImplementedError("This function is not implemented for float 8 as not needed.")
+
+    qrange = None
+
+    if reduce_range:
+        qrange = ONNX_INT_TYPE_REDUCED_RANGE.get(qType)
+    elif sym and qType in ONNX_INT_TYPE_SYMMETRIC_RANGE:
+        qrange = ONNX_INT_TYPE_SYMMETRIC_RANGE[qType]
+    else:
+        qrange = ONNX_INT_TYPE_RANGE.get(qType)
+
+    if not qrange:
+        raise ValueError(f"Unexpected data type {qType} requested. Only INT8 and UINT8 are supported.")
+
+    return qrange
+
+
+def dtype_to_name(dtype_mapping, dtype):
+    """Map data type and its string representation."""
+    return list(dtype_mapping.keys())[list(dtype_mapping.values()).index(dtype)]
+
+
+def _get_blob_size(group_size, has_zp):  # pragma: no cover
+    """Get blob_size.
+
+    Args:
+        group_size (int): how many elements share one scale/zp
+        has_zp (bool): whether zero_point is None
+    """
+    if version.Version(ort.__version__) > constants.ONNXRT1161_VERSION:
+        blob_size = group_size // 2
+    elif has_zp:
+        blob_size = group_size // 2 + 4 + 1
+    else:
+        blob_size = group_size // 2 + 4
+    return blob_size
+
+
+def make_matmul_weight_only_node(
```

Comment:
As we discussed, move these functions that are only used by woq algorithm to onc/algorithms/woq/utility/py
```python
from onnx_neural_compressor.algorithms import utility as algo_utils
from onnx_neural_compressor.algorithms.weight_only import utitlity as woq_utils
```

Reply:
discussed with Yi, functions for weight only are less than 10, there is no need to add another utility.py